sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -530,6 +537,101 @@ EncryptionProcessor getEncryptionProcessor() {
+                throw new IllegalArgumentException("Increment patch operation is not allowed for encrypted path");
+            }
+            else if (patchOperation instanceof PatchOperationCore) {
+                JsonNode objectNode = EncryptionUtils.getSimpleObjectMapper().valueToTree(((PatchOperationCore)patchOperation).getResource());
+                itemPatchOperationMono =
+                    encryptionProcessor.encryptPatchNode(objectNode, ((PatchOperationCore)patchOperation).getPath()).map(encryptedObjectNode -> {
+                        return new PatchOperationCore<>(
+                            patchOperation.getOperationType(),
+                            ((PatchOperationCore)patchOperation).getPath(),
+                            encryptedObjectNode
+                        );
+                    });
+                }
+            monoList.add(itemPatchOperationMono);
+        }
+        Mono<List<PatchOperation>> encryptedPatchOperationsListMono =
+            Flux.mergeSequential(monoList).collectList();
+        CosmosPatchItemRequestOptions finalRequestOptions = options;
+
+        CosmosPatchOperations encryptedCosmosPatchOperations = CosmosPatchOperations.create();
+
+        return encryptedPatchOperationsListMono.flatMap(patchOperations -> {
+            this.cosmosPatchOperationsAccessor.getPatchOperations(encryptedCosmosPatchOperations).addAll(patchOperations);
+            return patchItemInternalHelper(itemId, partitionKey, encryptedCosmosPatchOperations, finalRequestOptions,itemType, false);
+        });
+    }
+
+    private <T> Mono<CosmosItemResponse<T>> patchItemInternalHelper(String itemId,
+                                                                    PartitionKey partitionKey,
+                                                                    CosmosPatchOperations encryptedCosmosPatchOperations,
+                                                                    CosmosPatchItemRequestOptions requestOptions,
+                                                                    Class<T> itemType,
+                                                                    boolean isRetry) {
+
+        setRequestHeaders(requestOptions);
+        return this.container.patchItem(itemId, partitionKey, encryptedCosmosPatchOperations, requestOptions, itemType).publishOn(encryptionScheduler).
+            flatMap(cosmosItemResponse -> setByteArrayContent((CosmosItemResponse<byte[]>)cosmosItemResponse,
+                this.encryptionProcessor.decrypt(this.cosmosItemResponseBuilderAccessor.getByteArrayContent((CosmosItemResponse<byte[]>)cosmosItemResponse)))
+                .map(bytes -> this.responseFactory.createItemResponse((CosmosItemResponse<byte[]>) cosmosItemResponse,
+                    itemType))).onErrorResume(exception -> {
+                        if(!isRetry && exception instanceof CosmosException) {
+                            this.encryptionProcessor.getIsEncryptionSettingsInitDone().set(false);
```

Comment:
isIncorrectContainerRid(cosmosException) condition missing , please check other api

Reply:
Added the check